viewer.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf8 -*-
```

Comment:
This `viewer.py` is interesting. However I will do it by `html`. Maybe you can keep this file and make it a separated project?